driver/src/main/scala/api/MongoDriver.scala

```diff
@@ -301,22 +378,20 @@ class MongoDriver(
         ()
       }
 
-      /*
-      case CloseWithTimeout(timeout) =>
-        if (isEmpty) context.stop(self)
-        else context.become(closing(timeout))
-         */
-
       case Close => {
         logger.debug(s"[$supervisorName] Close the supervisor")
 
-        if (isEmpty) context.stop(self)
-        else context.become(closing(Duration.Zero))
+        if (isEmpty) {
+          context.stop(self)
+          sender() ! Closed
+        } else {
+          context.become(closing(mutable.Queue[ActorRef](sender)))
+          driver.connectionMonitors.values.foreach(_.close())
+        }
       }
     }
 
-    @deprecated("Will be private", "0.12.6")
```

Comment:
At least this should be kept deprecated, as it in anyway it should be part of the API. If there is still need for it would be renamed as a `private` function.

Reply:
Btw as this function won't be removed until next major release, this change is breaking compatibility. Have a look at MiMa reports.

Reply:
not sure why this was still deprecated with "Will be private", as the whole `SupervisorActor` class is already private anyway... but I will have a look a MiMa reports.

Reply:
Right, compatibility should not be an issue there.